.typos.toml

```diff
@@ -0,0 +1,21 @@
+default.extend-ignore-re = [
+  "/Ot",
+  "mis-alignment",
+  "mis-spelled",
+  "posix-seh-rt",
+]
+
+[default.extend-words]
+MAPE = "MAPE"
+datas = "datas"
+interprete = "interprete"
```

Comment:
LGTM except leaving "interprete"!
See https://github.com/microsoft/LightGBM/pull/6564#issuecomment-2248143004 for my proposal.

Reply:
Let's use this thread to discuss that proposal.

Please remember that breaking the public API of an R package is more difficult than doing it for a Python package, because of CRAN's restrictions.

CRAN will check the "strong reverse dependencies" (anything including `lightgbm` in `LinkingTo`, `Depends`, or `Imports`, I think) and check if a new submission breaks any of their builds or tests. This is why for the v4.0.0 release, I went around to all the packages listed at https://cran.r-project.org/web/packages/lightgbm/index.html as depending on `{lightgbm}` and made contributions to make them compatible with both v3.x and v4.x of `{lightgbm}`.

HOWEVER.... it does not look like any package on CRAN is using `lgb.interprete()`: https://github.com/search?q=org%3Acran%20%22lgb.interprete%22&type=code

So I'm ok with @StrikerRUS 's proposal, which is:

* introduce a new function `lgb.interpret()` which just calls `lgb.interprete()`
* raise a deprecation warning in direct calls to `lgb.interprete()`
* eventually remove `lgb.interprete()`

@borchero if you are not comfortable making those R changes (like updating the `NAMESPACE` file or writing roxygen comments), let me know and I can do this in a follow-up PR.

Reply:
> Please remember that breaking the public API of an R package is more difficult than doing it for a Python package, because of CRAN's restrictions.

Ah, to be honest, I forgot about this!

> HOWEVER.... it does not look like any package on CRAN is using ``lgb.interprete()``

Thank God! 😃

Reply:
Sorry I took forever to get back to this @jameslamb. As far as I can tell, everything related to `lgb.interprete` was already reverted; the only text changes are in test docs. I'd be happy if you could take over for @StrikerRUS's proposal to call `lgb.interprete()` from `lgb.interpret()` in a separate PR, I'm not familiar with R at all 😅

Reply:
Sure! Happy to.

Reply:
https://github.com/microsoft/LightGBM/issues/6732